src/mito2/src/region/opener.rs

```diff
@@ -402,6 +417,12 @@ pub(crate) async fn replay_memtable<S: LogStore>(
     region_write_ctx.set_next_entry_id(last_entry_id + 1);
     region_write_ctx.write_memtable();
 
+    if allow_stale_entries {
```

Comment:
Do we need to add a flag to record whether there is a stale log and only obsolete the WAL when has_stale_log is true?

Reply:
> Do we need to add a flag to record whether there is a stale log and only obsolete the WAL when has_stale_log is true?

Resolved in 5287c50a2d